pkg/search/postgres/store.go

```diff
@@ -468,4 +580,226 @@ func GloballyScopedUpsertChecker[T any, PT unmarshaler[T]](targetResource permis
 	}
 }
 
+func (s *GenericStore[T, PT]) doDBWalk(ctx context.Context, walkFn func(PT) error) error {
+	fetcher, closer, err := RunCursorQueryForSchema[T, PT](ctx, s.schema, search.EmptyQuery(), s.db)
+	if err != nil {
+		return err
+	}
+	defer closer()
+	for {
+		rows, err := fetcher(cursorBatchSize)
+		if err != nil {
+			return pgutils.ErrNilIfNoRows(err)
+		}
+		for _, data := range rows {
+			if err := walkFn(data); err != nil {
+				return err
+			}
+		}
+		if len(rows) != cursorBatchSize {
+			break
+		}
+	}
+	return nil
+}
+
+func (s *GenericStore[T, PT]) resetCache(ctx context.Context) {
+	if !s.useCache {
```

Comment:
Applies to all helper functions: either we call them unconditionally in the store code and handle the flag within each function respectively, or we don't handle them within the function and call them conditionally in the store code.

Reply:
With the extraction of the cache into a dedicated object (as suggested), these conditions get dropped.